contracts/token/psp1155/src/traits.rs

```diff
@@ -214,33 +189,58 @@ pub trait IPSP1155: PSP1155Storage {
         _operator: AccountId,
         _from: AccountId,
         _to: AccountId,
-        _ids: Vec<Id>,
-        _amounts: Vec<Balance>,
+        _ids_to_amounts: Vec<(Id, Balance)>,
     ) {
     }
 
-    fn _mint(&mut self, to: AccountId, id: Id, amount: Balance) {
+    fn _mint(&mut self, to: AccountId, id: Id, amount: Balance) -> Result<(), PSP1155Error> {
         let operator = Self::env().caller();
 
         assert!(!to.is_zero(), "{}", PSP1155Error::TransferToZeroAddress.as_ref());
 
-        self._before_token_transfer(&vec![id]);
+        self._before_token_transfer(&vec![id], ZERO_ADDRESS.into(), to);
         self._increase_receiver_balance(to, id, amount);
 
-        self._do_safe_transfer_acceptance_check(operator, ZERO_ADDRESS.into(), to, id, amount, Vec::new());
+        let result = self._do_safe_transfer_acceptance_check(operator, ZERO_ADDRESS.into(), to, id, amount, Vec::new());
```

Comment:
After `_do_safe_transfer_acceptance_check` you need to use `?;`=) Because we should return before emitting if result is error) But we will refactor that, okey for now